client/trino-client/src/main/java/io/trino/client/FixJsonDataUtils.java

```diff
@@ -51,27 +51,26 @@
 import static io.trino.client.ClientStandardTypes.VARCHAR;
 import static java.util.Collections.unmodifiableList;
 import static java.util.Objects.requireNonNull;
-import static java.util.stream.Collectors.toList;
 
 final class FixJsonDataUtils
 {
     private FixJsonDataUtils() {}
 
-    public static Iterable<List<Object>> fixData(List<Column> columns, Iterable<List<Object>> data)
+    public static Iterable<List<Object>> fixData(List<Column> columns, List<List<Object>> data)
     {
         if (data == null) {
             return null;
         }
         requireNonNull(columns, "columns is null");
-        List<ClientTypeSignature> signatures = columns.stream()
+        ClientTypeSignature[] signatures = columns.stream()
```

Comment:
Why is this necessary?

Reply:
It's not strictly speaking necessary, but we were already creating a new `List<ClientTypeSignature>` on each invokacation anyway, and the contents are accessed for each cell (each row, each column) in the output, so creating an array instead of a list is a minor efficiency improvement (it also avoids a check cast on each iteration incurred on the generic list implementation).